### ISSUE TITLE
False positive "Use of possibly uninitialized variable"

### ISSUE THREAD

**Issue by y-kotov** [intellij-rust/intellij-rust#9865]
## Environment

* **IntelliJ Rust plugin version:** 0.4.184.5057-222
* **Rust toolchain version:** 1.61.0 (fe5b13d68 2022-05-18) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.2.4 (CL-222.4345.21)
* **Operating system:** Windows 10 10.0
* **Macro expansion:** enabled

## Problem description
False positive "Use of possibly uninitialized variable":

![image](https://user-images.githubusercontent.com/112939978/206397939-f495e1f4-00a0-4cb8-aae3-3eb82c08fcd4.png)

The program compiles and runs successfully.

## Steps to reproduce

``` Rust
fn main() {
    let mut x: &[u8] = &[1,2,3];
    println!("x = {:?}", x);

    let first_part: &[u8];
    (first_part, x) = x.split_at(1);
    println!("First = {:?}", first_part);

    println!("x = {:?}", x);
}
```
